server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -929,13 +929,11 @@ static Settings aggregateIndexSettings(
          * that will be used to create this index.
          */
         shardLimitValidator.validateShardLimit(indexSettings, currentState);
-        if (indexSettings.getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true) == false) {
-            DEPRECATION_LOGGER.deprecate(
-                "soft_deletes_disabled",
-                "Creating indices with soft-deletes disabled is deprecated and will be removed in future OpenSearch versions. "
-                    + "Please do not specify value for setting [index.soft_deletes.enabled] of index ["
-                    + request.index()
-                    + "]."
+        if (IndexSettings.INDEX_SOFT_DELETES_SETTING.get(indexSettings) == false
+            && IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(indexSettings).onOrAfter(Version.V_2_0_0)) {
```

Comment:
I'm a bit confused...this is server code that will run inside a version 2.0.0+ instance of the service. Why is this check necessary? Why can't the check just be

```
if (!IndexSettings.INDEX_SOFT_DELETES_SETTING.get(indexSettings)) {
    throw ...
}
```

Taking it a bit further, why can't we just remove the `index.soft_deletes.enabled` setting all together so that it cannot be specified since the soft delete behavior is not configurable?

Reply:
@andrross afaik when migrating from let say 1.x to 2.x, you could still use "old" index from 1.x in 2.x: in this case the version of the index will be `1.x`, with all the settings implied. If we drop this check, the "old" indices would behave differently.

Reply:
I see. So we'll continue to support the "soft delete disabled behavior" after this change because it is possible to migrate an index from an older version with index.soft_deletes.enabled=false to a 2.0+ version?

Reply:
Right, this is my understanding

Reply:
Correct. We need to be able to support the parameter through 2.x for bwc purposes.